capa/features/common.py

```diff
@@ -232,9 +235,12 @@ def evaluate(self, ctx, short_circuit=True):
         else:
             return Result(False, _MatchedSubstring(self, {}), [])
 
+    def get_value_str(self) -> str:
+        return escape_string(self.value)
+
     def __str__(self):
         assert isinstance(self.value, str)
-        return "substring(%s)" % self.value
+        return "substring(%s)" % escape_string(self.value)
```

Comment:
seems like that embedded NULL should not be there, so seems like a bug, definitely

Reply:
i think this happens because the sample indexes into a longer string:

<img width="530" alt="image" src="https://user-images.githubusercontent.com/156560/211305317-ae270603-f452-4610-be91-fa0f3fdd0791.png">

whereas vivisect "helpfully" returns the length of the longer string.

Reply:
#1271